packages/server/dataloader/LocalCache.ts

```diff
@@ -0,0 +1,177 @@
+import ms from 'ms'
+import {RethinkTypes} from '../database/rethinkDriver'
+import RedisCache from './RedisCache'
+import {Doc, RType, RWrite, Updater} from './RethinkDBCache'
+
+const resolvedPromise = Promise.resolve()
+
+// every original read has 1 key, 1 callback and 1 entry in the cacheMap
+// every duplicate request has 1 cacheHit that, when called, resolves to the original promise in the cacheMap
+export default class LocalCache {
+  private cacheMap = {} as {[key: string]: {ts: number; promise: Promise<any>}}
+  private hasReadDispatched = true
+  private hasWriteDispatched = true
+  private keys = [] as string[]
+  private callbacks = [] as {resolve: (payload: any) => void; reject: (err: any) => void}[]
+  private cacheHits = [] as (() => void)[]
+  private writes = [] as (RWrite<any> & {resolve: (payload: any) => void})[]
+  private ttl = ms('1h')
```

Comment:
Can we make this a configuration/env value? Benefit is that for local development we can set it to a rather low value so we could test the caching locally?

Reply:
for sure, we could make the ttl an argument to the constructor, but since this is a singleton i figured it'd be just as easy to change it here, no strong preference so if you've got one feel free!

i try to keep the .env free of client variables, with the exception of 3rd party client keys like google, github, etc, since those vary based on environment.